homeassistant/components/mutesync/binary_sensor.py

```diff
@@ -0,0 +1,68 @@
+"""mütesync binary sensor entities."""
+import asyncio
+
+import aiohttp
+import async_timeout
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the mütesync button."""
+    client = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([MuteStatus(client)], True)
+
+
+class MuteStatus(BinarySensorEntity):
+    """Class to hold mütesync basic info."""
+
+    def __init__(self, client):
+        """Initialize binary sensor."""
+        self.client = client
+        self._status = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "Mute Status"
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the sensor."""
+        if self._status is None:
+            return None
+        return f"{self._status['user-id']}-mute"
+
+    @property
+    def available(self):
+        """Return if state is available from sensor."""
+        return self._status and self._status["in_meeting"]
```

Comment:
`available` reflects the state of the device is available by itself. For example, it should become unavailable when the device is unreachable.

With this suggested implementation, this would mean the device/entity is completely unavailable when I'm not in meeting? Is that correct?

Reply:
So @balloob and I were thinking of using `available` to mean whether or not there is a meeting going on. Are we not allowed to use `available` for that purpose? Is there some other concept we should use in `BinarySensor`?

Your suggested change would make it always return true whenever status is available, which is consistent with your suggested use of `available`. But we were hoping to use `available` for a different purpose.

If a meeting is going on, then the mute status becomes available, otherwise, it is not available. Does that make sense?

Reply:
It is basically hiding an additional state in a property that is meant to indicate there is a problem/disconnect with the device providing the entity.

So, 2 possible solutions in my opinion:

- Make it a Sensor, as that would allow for multiple states.
- Make multiple binary sensors (one for mute, other for "in-meeting" state.

Personally, I would take the latter. I do realize this would also add additional work